### PR TITLE
Enable POLAR_LOG & friends for the wasm library

### PR DESCRIPTION
**Pull request by gj**
PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

File: polar-core/src/vm.rs [hunk 1]
```diff
@@ -704,13 +726,25 @@ impl PolarVirtualMachine {
     }
 
     /// Print a message to the output stream.
```
Comment (plotnick) [inline polar-core/src/vm.rs:728]:
```suggestion
    /// Push or print a message to the output stream.
```

File: polar-core/src/vm.rs [hunk 2]
```diff
@@ -704,13 +726,25 @@ impl PolarVirtualMachine {
...
         let message = message.into();
         if self.polar_log_stderr {
             eprintln!("{}", message);
+        } else {
+            self.messages.push(MessageKind::Print, message);
         }
+    }
 
-        self.messages.push(MessageKind::Print, message);
+    /// Print a message to the output stream.
```
Comment (plotnick) [inline polar-core/src/vm.rs:739]:
```suggestion
    /// Push or print a message to the WASM output stream.
```

File: polar-core/src/polar.rs [hunk 3]
```diff
@@ -262,12 +269,9 @@ impl Polar {
...
                 parser::parse_query(src_id, src).map_err(|e| e.set_context(Some(&source), None))?;
             kb.sources.add_source(source, src_id);
-            rewrite_term(term, &mut kb)
+            term
         };
-        let query = Goal::Query { term: term.clone() };
-        let vm =
-            PolarVirtualMachine::new(self.kb.clone(), trace, vec![query], self.messages.clone());
-        Ok(Query::new(vm, term))
+        Ok(self.new_query_from_term(term, trace))
```
Comment (plotnick) [inline polar-core/src/polar.rs:274]: Odd that it wasn't like this before.

**Review by plotnick — APPROVED** [review]
Looks good modulo :paperclip:. The three boolean arguments to the Polar constructor should be revisited in the fullness of time, but aren't blocking.

**Review by samscott89 — APPROVED** [review]
lgtm

**Pull request merged**